modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -113,6 +122,10 @@ class S3BlobStore implements BlobStore {
         this.snapshotExecutor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
         this.meterRegistry = meterRegistry;
         this.requestCounter = this.meterRegistry.getLongCounter(METRIC_REQUESTS_COUNT);
+        this.exceptionCounter = this.meterRegistry.getLongCounter(METRIC_EXCEPTIONS_COUNT);
+        this.throttleCounter = this.meterRegistry.getLongCounter(METRIC_THROTTLES_COUNT);
+        this.exceptionHistogram = this.meterRegistry.getLongHistogram(METRIC_EXCEPTIONS_HISTOGRAM);
+        this.throttleHistogram = this.meterRegistry.getLongHistogram(METRIC_EXCEPTIONS_HISTOGRAM);
```

Comment:
This list is getting long. I'd like to replace them with a dedicate RepositoryMetrics class. It requires a few changes for wiring it properly. So I'll address it in a separate PR.

Reply:
FWIW, here's an [example of what we've been doing](https://github.com/elastic/elasticsearch/blob/e452fbd8f5ec19b61c900054bec13d2e83d67074/x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java), for [this reason](https://github.com/elastic/elasticsearch/pull/101577/files#r1379132909).